packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmListInternal.kt

```diff
@@ -133,13 +134,15 @@ internal class ManagedRealmList<E>(
 
     override fun freeze(frozenRealm: RealmReference): ManagedRealmList<E>? {
         return RealmInterop.realm_list_resolve_in(nativePointer, frozenRealm.dbPointer)?.let {
-            ManagedRealmList(it, metadata.copy(realm = frozenRealm))
+            val converter: RealmValueConverter<E> = converter<Any>(metadata.clazz, metadata.mediator, frozenRealm) as CompositeConverter<E, *>
+            ManagedRealmList(it, metadata.copy(realm = frozenRealm, converter = converter))
         }
     }
 
     override fun thaw(liveRealm: RealmReference): ManagedRealmList<E>? {
         return RealmInterop.realm_list_resolve_in(nativePointer, liveRealm.dbPointer)?.let {
-            ManagedRealmList(it, metadata.copy(realm = liveRealm))
+            val converter: RealmValueConverter<E> = converter<Any>(metadata.clazz, metadata.mediator, liveRealm) as CompositeConverter<E, *>
+            ManagedRealmList(it, metadata.copy(realm = liveRealm, converter = converter))
```

Comment:
Yeah. I have already changed it to an explicit interface method in https://github.com/realm/realm-kotlin/pull/797/files#diff-6ae484901f6301d8cce7b3b571876dbdefc491334d69fe44ec1f77ce69a96acfR212, but since this is a hot-fix I just added it to the existing data class's copy method. We are only copying the metadata when thawing/freezing so think it should be ok for the time being and until #797 is merged.